utility/src/main/java/org/oppia/util/profile/DirectoryManagementUtil.kt

```diff
@@ -25,6 +26,16 @@ class DirectoryManagementUtil @Inject constructor() {
    * @return whether directory was successfully deleted.
    */
   fun deleteDir(profileId: String): Boolean {
-    return true
+    return deleteRecursive(getOrCreateDir(profileId))
+  }
+
+  /** Directories must be empty first before deleting */
+  private fun deleteRecursive(file: File): Boolean {
```

Comment:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/delete-recursively.html :)

(Kotlin usually solves things that Java doesn't; I keep finding things like this just by searching for the algorithm + 'Kotlin').